### PR TITLE
Ouput CSV files with the csv file extension

### LINKED ISSUE #9044: Output files with .csv file extension instead of .part when to_csv is used
The `to_csv` method outputs filenames with a `.part` extension by default.  This post argues that `to_csv` should output CSV files with a `.csv` extension by default.

Let's create a DataFrame and write it out to a directory and see the default behavior.

```python
import pandas as pd
import dask.dataframe as dd

pdf = pd.DataFrame(
    {"num1": [1, 2, 3, 4], "num2": [7, 8, 9, 10]},
)

ddf = dd.from_pandas(pdf, npartitions=2)

ddf.to_csv("./data/csv_simple")
```

Here are the files that are output:

```
csv_simple/
  0.part
  1.part
```

Downstream readers need to do something like `dd.read_csv("./data/csv_simple/*.part")` to read these CSV files into a DataFrame.

Here's the work-around to output the files with the `.csv` file extension:

```python
ddf.to_csv("./data/csv_simple2/whatever-*-hi.csv")
```

Here are the files that are output:

```
csv_simple2/
  whatever-0-hi.csv
  whatever-1-hi.csv
```

I don't believe there is a work around to customize the file extension and avoid writing `.part` when you use the `name_function` argument.  Take a look at this example:

```python
ddf.to_csv("./data/csv_simple3", name_function = lambda x: f"i-like-{x}.csv")
```

Here's what's output:

```
csv_simple3/
  i-like-0.csv.part
  i-like-1.csv.part
```

I'd personally prefer for files to be written with a `.csv` extension by default.  That'd be more intuitive for me.

Think it's also more consistent with the Parquet writers.  For example `ddf.to_parquet("./data/parquet_simple")` outputs files like `part.0.parquet` and `part.1.parquet`.  Let me know what you think!

### PR DESCRIPTION
**Pull request by MrPowers**
- [X] Closes #9044 
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

This PR causes Dask to output files like `part-1.csv` and `part-2.csv` by default instead of `0.part` and `1.part` when `ddf.to_csv("some/folder")` is run.  This PR should not change the behavior when the user specifies a globstring like `ddf.to_csv("some/folder/whatever-*.csv")`.

Users that still want the `.part` file extension can run `ddf.to_csv("some/folder/*.part")`.  They'll need to update their code, but this can be made backwards compatible for legacy systems.

The implementation in this PR was suggested by @martindurant in [this issue](https://github.com/dask/dask/issues/9044): "the quick fix for dask is to add its own "*" into the path".

Feel free to leave thoughts / suggestions.  The high level motivation for this PR is that I'm writing blog posts / a Dask book and the default behavior is hard to explain.  Users are confused that their CSV files aren't being output with the right file extension.  Hopefully this new default file extension is more intuitive.

### CODE REVIEW

**Comment by MrPowers** [comment]
@rjzamora - thanks for reviewing and agree with your user-friendly messaging.  Let's see what James thinks.  

@martindurant - feel free to provide feedback too if you have any thoughts!

**Comment by martindurant** [comment]
All good from me

File: dask/dataframe/io/csv.py [hunk 1]
```diff
@@ -915,6 +915,8 @@ def to_csv(
         values = [value]
         files = [first_file]
     else:
+        if os.path.isdir(filename) and "*" not in filename:
+            filename = os.path.join(filename, "part-*.csv")
```
Comment (jcrist) [inline dask/dataframe/io/csv.py:919]: `to_parquet` outputs files like `part.{index}.parquet`, perhaps we should do the same here for consistency?

```suggestion
            filename = os.path.join(filename, "part.*.csv")
```

**Comment by MrPowers** [comment]
@jcrist - thanks for taking a look.  Updated the code to output a filename that's consistent with what `to_parquet` outputs - good catch!

File: dask/dataframe/io/csv.py [hunk 2]
```diff
@@ -915,6 +915,8 @@ def to_csv(
         values = [value]
         files = [first_file]
     else:
+        if os.path.isdir(filename) and "*" not in filename:
```
Comment (ian-r-rose) [inline dask/dataframe/io/csv.py:918]: A suggestion: most Dask I/O functions use `fsspec` as an abstract interface to file-system-like storage backends. This is what allows us to easily write to things like cloud storage using the same APIs as local filesystems.

`os.path.isdir` will *only* work for the local filesystem. Instead, I suggest using something like `fsspec.get_fs_token_paths` or `fsspec.url_to_fs`:

```python
fs = fsspec.url_to_fs(filename)
if fs.isdir(filename) and "*" not in filename:
    filename = fs.sep.join(filename, "part.*.csv")
```

**Review by ian-r-rose — COMMENTED** [review]
Sorry to let this languish @MrPowers! 

Do you have thoughts about whether to implement @rjzamora's suggested user warning about the change?

**Comment by pavithraes** [comment]
@MrPowers Thanks for this PR! What do you think about `@ian-r-rose`'s comments?

**Pull request closed without merging**